packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -598,28 +598,82 @@ describe("SharedTree", () => {
 
 			const log: string[] = [];
 			const unsubscribe = root[on]("changing", () => log.push("change"));
+			const unsubscribeSubtree = root[on]("subtreeChanging", () => log.push("subtree"));
 			const unsubscribeAfter = tree1.events.on("afterBatch", () => log.push("after"));
 			log.push("editStart");
 			root[valueSymbol] = 5;
 			log.push("editStart");
 			root[valueSymbol] = 6;
 			log.push("unsubscribe");
 			unsubscribe();
+			unsubscribeSubtree();
 			unsubscribeAfter();
 			log.push("editStart");
 			root[valueSymbol] = 7;
 
 			assert.deepEqual(log, [
 				"editStart",
+				"subtree",
 				"change",
 				"after",
 				"editStart",
+				"subtree",
 				"change",
 				"after",
 				"unsubscribe",
 				"editStart",
 			]);
 		});
+
+		it("propagates path and value args for local and subtree changes", async () => {
+			const value = "42";
+			const provider = await TestTreeProvider.create(1);
```

Comment:
Since this test does not need support for summarization, you can use the simpler and faster MockFluidDataStoreRuntime approach to making a test tree, see src/test/shared-tree/schmatizeTree.spec.ts for some examples.

That file also use the new schematizeView API, and typed schema apis, which would be preferred here if you are going to setup an actual shared-tree.

Reply:
I know a lot of our tests use the patterns you used here: we are trying to migrate away from them.

Reply:
Understood. Thanks for the pointers!

Reply:
Modified both event tests to use `MockFluidDataStoreRuntime` and `schematizeView` APIs.